src/Modal.js

```diff
@@ -277,6 +277,11 @@ const Modal = React.createClass({
     }
   },
 
+  componentWillUnmount() {
+    this.onHide();
+    this.props.onHide();
```

Comment:
@jquense 

> because the modal isn't being hidden it is being unmounted. 

when it's unmounted it's being hidden without doubts (if it's shown, that's what I missed indeed)

> not calling the onHide prop callback is consistent with every the other components, and with the react ecosystem in general (what I am familiar with of course)

That is arguable: I as a developer and consumer of `Modal` component expect `onHide` to be invoked when the component disappears, irregardless on what actually caused that.